lib/wappalyzer.rb

```diff
@@ -17,12 +18,24 @@ def initialize
       @categories, @apps = @json['categories'], @json['apps']
     end
 
+    def utf8_encoding(str)
+      str.encode('UTF-8', :invalid => :replace, :undef => :replace)
+    end
+
     def analyze(url)
       uri, body, headers = URI(url), nil, {}
-      Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https', :open_timeout => 5) do |http|
-        resp = http.get(uri.request_uri)
-        resp.each_header{|k,v| headers[k.downcase] = v}
-        body = resp.body.encode('UTF-8', :invalid => :replace, :undef => :replace)
+      Net::HTTP.start(uri.host, uri.port,
+                      :use_ssl => uri.scheme == 'https',
+                      :verify_mode => OpenSSL::SSL::VERIFY_NONE,
+                      :open_timeout => 5) do |http|
+        begin
+          resp = http.get(uri.request_uri)
+        rescue Zlib::DataError
+          resp = http.get(uri.request_uri, "Accept-Encoding" => "none")
+        end
+
+        resp.each_header { |k,v| headers[k.downcase] = utf8_encoding(v) }
```

Comment:
To be 100% safe, I think that's a fine idea!

I believe all possible header keys are listed [here](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields), and they all look like they use basic utf-8 characters, so if there's a performance hit from doing this it can probably be reversed without error.